x/mongo/driver/topology/sdam_spec_test.go

```diff
@@ -304,147 +304,12 @@ func (n netErr) Temporary() bool {
-		case "errmsg":
-			if str, okay := elem.Value().StringValueOK(); okay {
-				errmsg = str
-			}
-		case "codeName":
-			if str, okay := elem.Value().StringValueOK(); okay {
-				codeName = str
-			}
-		case "code":
-			if c, okay := elem.Value().Int32OK(); okay {
-				code = c
-			}
-		case "errorLabels":
-			if arr, okay := elem.Value().ArrayOK(); okay {
-				elems, err := arr.Elements()
-				if err != nil {
-					continue
-				}
-				for _, elem := range elems {
-					if str, ok := elem.Value().StringValueOK(); ok {
-						labels = append(labels, str)
-					}
-				}
-
-			}
-		case "writeErrors":
-			arr, exists := elem.Value().ArrayOK()
-			if !exists {
-				break
-			}
-			vals, err := arr.Values()
-			if err != nil {
-				continue
-			}
-			for _, val := range vals {
-				var we driver.WriteError
-				doc, exists := val.DocumentOK()
-				if !exists {
-					continue
-				}
-				if index, exists := doc.Lookup("index").AsInt64OK(); exists {
-					we.Index = index
-				}
-				if code, exists := doc.Lookup("code").AsInt64OK(); exists {
-					we.Code = code
-				}
-				if msg, exists := doc.Lookup("errmsg").StringValueOK(); exists {
-					we.Message = msg
-				}
-				wcError.WriteErrors = append(wcError.WriteErrors, we)
-			}
-		case "writeConcernError":
-			doc, exists := elem.Value().DocumentOK()
-			if !exists {
-				break
-			}
-			wcError.WriteConcernError = new(driver.WriteConcernError)
-			if code, exists := doc.Lookup("code").AsInt64OK(); exists {
-				wcError.WriteConcernError.Code = code
-			}
-			if name, exists := doc.Lookup("codeName").StringValueOK(); exists {
-				wcError.WriteConcernError.Name = name
-			}
-			if msg, exists := doc.Lookup("errmsg").StringValueOK(); exists {
-				wcError.WriteConcernError.Message = msg
-			}
-			if info, exists := doc.Lookup("errInfo").DocumentOK(); exists {
-				wcError.WriteConcernError.Details = make([]byte, len(info))
-				copy(wcError.WriteConcernError.Details, info)
-			}
-			if errLabels, exists := doc.Lookup("errorLabels").ArrayOK(); exists {
-				elems, err := errLabels.Elements()
-				if err != nil {
-					continue
-				}
-				for _, elem := range elems {
-					if str, ok := elem.Value().StringValueOK(); ok {
-						labels = append(labels, str)
-					}
-				}
-			}
-		case "topologyVersion":
-			doc, ok := elem.Value().DocumentOK()
-			if !ok {
-				break
-			}
-			version, err := description.NewTopologyVersion(bson.Raw(doc))
-			if err == nil {
-				tv = version
-			}
-		}
-	}
-
-	if errmsg == "" {
-		errmsg = "command failed"
-	}
-
-	return driver.Error{
-		Code:            code,
-		Message:         errmsg,
-		Name:            codeName,
-		Labels:          labels,
-		TopologyVersion: tv,
-	}
-}
-
 func applyErrors(t *testing.T, topo *Topology, errors []applicationError) {
 	for _, appErr := range errors {
 		var currError error
 		switch appErr.Type {
 		case "command":
-			currError = getError(appErr.Response)
+			currError = driver.ExtractErrorFromServerResponse(appErr.Response)
```

Comment:
Ah nice, using the newly exported `ExtractErrorFromServerResponse` seems so much better than using `getError` copy.